Own player implementation

df530 opened the pull request.

df530 committed:
Add own player and fix some bags

Rewind and forward increment implemented with buttons.
Also added lock rotation button and progress bar

df530 committed:
Add double tap 10 seconds increment

DF5HSE commented:
https://trello.com/c/unJ0IMTI

df530 committed:
Add playing youtube video and update progress bar

df530 committed:
Fix lock button changing color

df530 committed:
Merge branch 'main' into media-player

# Conflicts:
#	app/build.gradle
#	build.gradle

df530 committed:
Add speed change by seek bar and by double tap

DF5HSE commented:
Код ужасный, пока можешь особо не смотреть, я в четверг, скорее всего сяду и буду все поправлять, в том числе то, о чем мы с тобой на созвоне говорили

DF5HSE commented:
https://trello.com/c/jQ9SeB5V

df530 committed:
Hide title bar in all focus changes

When you opened setting curtain or connect to PC
by USB, you could see title bar. Now you can't

df530 committed:
Add show new speed after double tap

df530 committed:
Change buffer params to decr video load time. Fix vertical seek bar thumb location

df530 committed:
Add set play/pause by headphones

df530 committed:
Implement model-view pattern with RX-java and change buffer params

df530 committed:
Add foreground service with notification player

Now it write video name and author (for youtube) / URI (others).
TODO: add preview to the service and timebar

tilacyn commented on app/src/main/java/com/example/avp/player/AVPMediaMetaData.java:
```diff
@@ -0,0 +1,31 @@
+package com.example.avp.player;
+
+public class AVPMediaMetaData {
```
можно использовать lombok, пример https://github.com/DF5HSE/AndroidVideoPlayer/pull/7/files#diff-06f15bcbad8ff9df19a4707b62972254943d2238e37c330e6c6e683787c58e23

tilacyn commented on app/src/main/java/com/example/avp/player/ExoPlayerActivity.java:
```diff
@@ -1,94 +1,234 @@
+    }
+
+    private void initControllerElements() {
+        player.addListener(new Player.EventListener() {
```
Очень большой метод, надо бы его разбить, можно классы выделить вместо анонимных

tilacyn commented on app/src/main/java/com/example/avp/player/ExoPlayerModel.java:
```diff
@@ -0,0 +1,128 @@
+
+    public Observable<MediaSource> getMediaSource() {
+        Observable<MediaSource> res;
+        if (isYoutubeUrl(linkOnVideo)) {
```
этот разбор случаев можно переделать через интерфейсы

tilacyn commented on app/src/main/java/com/example/avp/player/ExoPlayerModel.java:
```diff
@@ -0,0 +1,128 @@
+
+    private Observable<MediaSource> getObservableMediaSourceFromUri() {
+        AVPMediaMetaData meta = new AVPMediaMetaData(new File(linkOnVideo).getName(), null, linkOnVideo, null);
+        return Observable.fromCallable(() ->
```
Здесь fromCallable лучше заменить на just

tilacyn commented on app/src/main/java/com/example/avp/player/ExoPlayerModel.java:
```diff
@@ -0,0 +1,128 @@
+    @SuppressLint("StaticFieldLeak")
+    private Observable<MediaSource> getObservableMediaSourceFromYouTube() {
+        PublishSubject<MediaSource> resObservable = PublishSubject.create(); // create a list to save result MediaSource
+        new YouTubeExtractor(context) {
```
Этот анонимный класс точно нужно в отдельный класс (заодно можно реализовать интерфейс который возвращает observable)

tilacyn commented on app/src/main/java/com/example/avp/player/HeadsetButtonReceiver.java:
```diff
@@ -0,0 +1,78 @@
+        }
+    };
+
+    interface onHeadsetListener{
```
интерфейсы и классы надо именовать с большой буквы

df530 committed:
Add loading video preview and saving in metadata

df530 committed:
Add time seek bar to notification and fix some comments

df530 committed:
Add restarting notification, if it was deleted, but video continues

vladnosiv committed:
Merge branch 'main' into media-player

# Conflicts:
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/example/avp/player/ExoPlayerActivity.java
#	app/src/main/res/values/strings.xml

vladnosiv committed:
integrate gdrive code with new exoplayermodel
